### PR TITLE
Virtualbox cloud vendor option

### PR DESCRIPTION
**Pull request by vincentp7212**
Added Virtualbox option to postgres-deployment (Using Vagrant instead of mikemech to provision)

### CODE REVIEW

File: edbdeploy/data/virtualbox/c8-EDB-RA-3 [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    end
+
+    (1..N).each do |i|
+        config.vm.define "standby-#{i}" do |standby|
+            standby.vm.box = IMAGE_NAME
+            standby.vm.network "private_network", ip: STARTING_IP + "#{STANDBY_STARTING_IP_COUNT + i}"
+            standby.vm.hostname = STANDBY_BOX_NAME + "-#{i}"
+        end
+    end
+end
```
Comment (jt-edb) [inline edbdeploy/data/virtualbox/c8-EDB-RA-3:68]: I don't see anything related to pgpool. 3 Pgpool nodes are part of the EDB-RA-3 reference architecture.

File: edbdeploy/data/virtualbox/playbook.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
```
Comment (jt-edb) [inline edbdeploy/data/virtualbox/playbook.yml:1]: Do we need a specific ansible playbook for virtualbox? Please see https://github.com/EnterpriseDB/postgres-deployment/tree/master/edbdeploy/data/ansible

File: edbdeploy/projects/virtualbox.py [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+
+class VirtualBoxProject(Project):
+    def __init__(self, name, env, bin_path=None):
+        super(VirtualBoxProject, self).__init__('virtualbox', name, env, bin_path)
+
+    def hook_post_configure(self, env):
+        # Hook function called by Project.configure()
+        # Build the vars files for Ansible
+        self._build_ansible_vars_file(env)
+        # Copy VirutalBox Vagrant Config File into project dir.
```
Comment (jt-edb) [inline edbdeploy/projects/virtualbox.py:24]: VirtualBox

File: edbdeploy/projects/virtualbox.py [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+        mem_size = self.ansible_vars['mem_size']
+        cpu_count = self.ansible_vars['cpu_count']
+        vagrant = VirtualBoxCli(
+            self.cloud, self.name, self.cloud, mem_size, cpu_count,
+            self.vagrant_project_path, bin_path=self.cloud_tools_bin_path
+        )
+        with AM("Checking instances availability"):
+            vagrant.up()
+        # Build ip address list for virtualbox deployment
+        with AM("Build VirutalBox Ansible IP addresses"):
```
Comment (jt-edb) [inline edbdeploy/projects/virtualbox.py:44]: VirtualBox

File: edbdeploy/projects/virtualbox.py [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+            'virtualbox',
+            self.name,
+        )
+        mem_size = self.ansible_vars['mem_size']
+        cpu_count = self.ansible_vars['cpu_count']
+        vagrant = VirtualBoxCli(
+            self.cloud, self.name, self.cloud, mem_size, cpu_count,
+            self.vagrant_project_path, bin_path=self.cloud_tools_bin_path
+        )
+        with AM("Checking instances availability"):
```
Comment (jt-edb) [inline edbdeploy/projects/virtualbox.py:115]: Maybe we could update action's name here?

File: edbdeploy/projects/virtualbox.py [hunk 6]
```diff
@@ -0,0 +1,505 @@
...
+            vagrant.up()
+
+        # Build ip address list for VirutalBox deployment
+        with AM("Build VirutalBox Ansible IP addresses"):
```
Comment (jt-edb) [inline edbdeploy/projects/virtualbox.py:119]: VirtualBox

File: edbdeploy/projects/virtualbox.py [hunk 7]
```diff
@@ -0,0 +1,505 @@
...
+                    env.cloud_spec['pooler_server_%s' % i]['public_ip'] = result[0]  # noqa
+                    env.cloud_spec['pooler_server_%s' % i]['private_ip'] = result[0]  # noqa
+                except Exception as e:
+                    logging.error("Failed to execute the command")
+                    logging.error(e)
+                    raise CliError(
+                        ("Failed to obtain VirutalBox Instance IP Address for: %s,"
+                         "please check the logs for details.")
+                        % env.cloud_spec['pooler_server_%s' % i]['name']
+                    )
```
Comment (jt-edb) [inline edbdeploy/projects/virtualbox.py:388]: I'd suggest you to rewrite this function by generating the machine list based on the reference architecture, then, iterate over each machine name and execute the `vagrant ip <machine-name>` command.

**Review by jt-edb — COMMENTED** [review]
Please take a look at my comments. Thank you!

File: edbdeploy/projects/virtualbox.py (hunk 5 above)
Comment (vincentp7212) [reply edbdeploy/projects/virtualbox.py:115]: Changed it to `Provisioning Virtual Machines`

File: edbdeploy/data/virtualbox/playbook.yml (hunk 2 above)
Comment (vincentp7212) [reply edbdeploy/data/virtualbox/playbook.yml:1]: I just used the same playbook that was used for vmware-wkstn

Comment (jt-edb) [reply edbdeploy/data/virtualbox/playbook.yml:1]: Please use the `EDB-RA-1.yml`, `EDB-RA-2.yml` and `EDB-RA-3.yml` playbook files, located in `data/ansible`.

Comment (vincentp7212) [reply edbdeploy/data/virtualbox/playbook.yml]: Okay

**Review by jt-edb — APPROVED** [review]
LGTM

**Pull request merged**